Add an option to Plugin Assets Symlink for a relative path. Has testcase.

umer936 opened the pull request:
I needed this for cases where asset symlinks become wrong if they were generated inside a container but it's volume mounted as well as when I move an application between systems.

umer936 committed:
Add an option to Plugin Assets Symlink for a relative path

Also adds testcase

umer936 committed:
Update PluginAssetsTrait.php

Fix Coding Standards/Static Analysis

umer936 committed:
Continue fixing Code Standards

umer936 committed:
Code Checker

umer936 committed:
Use "use" for namespace in PluginAssetsTrait

umer936 commented:
I don't actually know how to make a test to test the `InvalidArgumentException` in https://github.com/cakephp/cakephp/pull/18446/files#annotation_34304084181, so if someone could do that for me please

also the Windows thing

markstory commented on tests/TestCase/Command/PluginAssetsCommandsTest.php:
```diff
@@ -91,6 +91,29 @@ public function testSymlink(): void
+     */
+    public function testRelativeSymlink(): void
+    {
+        $this->loadPlugins(['TestPlugin' => ['routes' => false], 'Company/TestPluginThree']);
```
We may need to skip this test on windows because of reduced symlink support.

```suggestion
        $this->skipIf(DS === '\\', 'Cant perform operations with symlinks windows.');
        $this->loadPlugins(['TestPlugin' => ['routes' => false], 'Company/TestPluginThree']);
```

ADmad commented:
Given the addition of the new option and the method signature changes this should target `5.next`.

umer936 committed:
Update tests/TestCase/Command/PluginAssetsCommandsTest.php

Skip Plugin Assets Relative Symlink test on Windows as it does not support symlinks as is. Can do symlinks with admin privileges and such but likely better to skip it then try to admin for making symlinks.

Co-authored-by: Mark Story <mark@mark-story.com>

umer936 commented:
Did something fancy to base it off 5.next while keeping the history and not cluttering the "Files Changed" with the merge or rebase.

As for the Windows, I did Mark's suggestion. There's ways to do symlinks on Windows but not in ways we'd want people to do I would imagine.
